### PR TITLE
New property rule category - "formulas"

### PR DESCRIPTION
**Pull request by emhomm4**
This PR creates a new PropertyRuleCategory for named formulas, called "Formulas".

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/App/Controls/PropertyRuleCategory.cs [hunk 1]
```diff
@@ -21,12 +21,13 @@ internal enum PropertyRuleCategory
...
         OnDemandData = 4,
         Scope = 5,
+        Formulas = 6,
         /// <summary>
         /// Represents a missing property category when deserializing
         /// Should be cleaned up by document converter, only occurs if 
         /// control template is invalid
         /// </summary>
-        Unknown = 6,
+        Unknown = 7,
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/App/Controls/PropertyRuleCategory.cs:30]: why are we renumbering this enum?

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Core/App/Controls/PropertyRuleCategory.cs:30]: IE, should `Unknown` be the last item?  Or should Formulas be added as 7?

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/App/Controls/PropertyRuleCategory.cs:30]: Unknown is a valid value, it's not a catch-all, and it's serialized by PowerApps-Client. It should stay 6, the new one should be added as 7

**Review by lesaltzm — APPROVED** [review]
🚢

**Pull request merged**